sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileClient.java

```diff
@@ -0,0 +1,375 @@
+     * @return A response containing information about the created file
+     */
+    public Response<PathInfo> createWithResponse(PathHttpHeaders headers, Map<String, String> metadata,
+        DataLakeRequestConditions accessConditions, String permissions, String umask, Duration timeout,
+        Context context) {
+        Mono<Response<PathInfo>> response = pathAsyncClient.createWithResponse(PathResourceType.FILE, headers, metadata,
+            accessConditions, permissions, umask, context);
+
+        return StorageImplUtils.blockWithOptionalTimeout(response, timeout);
+    }
+
+    /**
+     * Deletes a file.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.file.datalake.DataLakeFileClient.delete}
+     *
+     * <p>For more information see the
+     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/delete">Azure
+     * Docs</a></p>
+     */
+    public void delete() {
+        deleteWithResponse(null, null, Context.NONE).getValue();
+    }
+
+    /**
+     * Deletes a file.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.file.datalake.DataLakeFileClient.deleteWithResponse#DataLakeRequestConditions-Duration-Context}
+     *
+     * <p>For more information see the
+     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/delete">Azure
+     * Docs</a></p>
+     *
+     * @param accessConditions {@link DataLakeRequestConditions}
+     * @param timeout An optional timeout value beyond which a {@link RuntimeException} will be raised.
+     * @param context Additional context that is passed through the Http pipeline during the service call.
+     *
+     * @return A response containing status code and HTTP headers.
+     */
+    public Response<Void> deleteWithResponse(DataLakeRequestConditions accessConditions, Duration timeout,
+        Context context) {
+        // TODO (rickle-msft): Update for continuation token if we support HNS off
+        Mono<Response<Void>> response = pathAsyncClient.deleteWithResponse(null, accessConditions, context);
+
+        return StorageImplUtils.blockWithOptionalTimeout(response, timeout);
+    }
+
+    /**
+     * Appends data to the specified resource to later be flushed (written) by a call to flush
+     *
+     * <p><strong>Code Samples>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.file.datalake.DataLakeFileClient.append#InputStream-long-long}
+     *
+     * <p>For more information, see the
+     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/update">Azure
+     * Docs</a></p>
+     *
+     * @param data The data to write to the file.
+     * @param offset The position where the data is to be appended.
+     * @param length The exact length of the data.
+     */
+    public void append(InputStream data, long offset, long length) {
```

Comment:
>long offset [](start = 41, length = 11)

I think sometimes offset parameters in methods like this indicate where to start reading from the source and not where to start writing into the destination. In this case, I propose we rename this parameter fileOffset for clarity

Reply:
Done